sbi/inference/snpe/snpe_a.py

```diff
@@ -544,17 +547,21 @@ def _posthoc_correction(self, x: Tensor):
         estimator and the proposal.
 
         Args:
-            x: Conditioning context for posterior.
+            x: Conditioning context for posterior, shape
+                `(batch_dim, *event_shape)`.
 
         Returns:
             Mixture components of the posterior.
         """
+        # Remove the batch dimension of `x` (SNPE-A always has a single `x`).
+        x = x.squeeze(dim=0)
 
         # Evaluate the density estimator.
         embedded_x = self._neural_net.net._embedding_net(x)
         dist = self._neural_net.net._distribution  # defined to avoid black formatting.
         logits_d, m_d, prec_d, _, _ = dist.get_mixture_components(embedded_x)
         norm_logits_d = logits_d - torch.logsumexp(logits_d, dim=-1, keepdim=True)
+        norm_logits_d = atleast_2d(norm_logits_d)
```

Comment:
why do we need this new dimensions here?

Reply:
This was rather a bugfix. The bug never surfaced because torch does some magic broadcasting as part of the `repeat_interleave` in `_logits_posterior`.